### DISCUSSION TITLE
settings/parameters already provided by spectral library

### DISCUSSION THREAD

**Discussion by JM-Bader** [vdemichev/DiaNN#402] [Q&A] [answered]
One can use DIANN in library-free and library mode. 
Some features of the search (e.g. PTMs, charge states, peptide length, missed cleavage) should already be specified by the spectral library. 

Q1: What is the impact of setting these in DIANN in a library-based search?
:E.g. What happens when the spectral library 
- contains 2 miscleavages (default in MSFragger)
- has a peptide length 7-50 (default in Spectronaut)
- has various PTMs (e.g. Ox(M), Ac(Nterm)) 
and differing settings are applied in DIANN ?
- e.g 1 miscleavage
- peptide length 7-30
- disabled ox(M) / disabled Ac(Nterm)

Q2: The other way round, is it a problem if DIANN settings are more relaxed than the spectral library? 
- for isntance DIANN charge state 1-4, library 2-3

Q3: - or MSFragger lib has filter : max fragment charge 2. (DIANN does not appear to have such a filter). >> Should one relaxe this for MSFragger 

Q5: What is the impact of providing a fasta for the library based search when protein inference is off?

Q6: 
![image](https://user-images.githubusercontent.com/98888664/171867601-f4369ad0-681a-4eb6-8539-bbe46446d262.png)
What does 'in general no' mean?
What does it mean that the PTMs in the spectral library do not need to be recognised ? Naively, one would think this is important for peptide / fragment identification. (Or what is the difference to 'recognizing the modifications it searches' ?)

**Comment by vdemichev** [comment, marked as answer]
Q1-Q2. Precursor generation settings only have impact on precursor generation, e.g. when DIA-NN digest a FASTA database in silico. Effect when using a spectral library only in specific cases, e.g. when using Reannotate or when enabling PTM scoring (on by default for N-acetyl, phospho and ubiquitin GUI options).
Q3. In silico prediction in DIA-NN likewise only considers fragment charges 1 or 2.
Q5. Gene names in the library are replaced with those from FASTA, by matching protein sequence IDs to FASTA.
Q6. The screenshot answers that. Indeed, for peptide identification it's enough to know fragment masses, aminoa cid composition is not particularly relevant.
